woocommerce/wc-smooth-generator
# Remove Grunt and related dependencies, run phpcs from composer

### Changes proposed in this Pull Request:

The Gruntfile has two commands, one for running the code sniffer (phpcs) and one for checking the text domain on gettext strings. The latter is not necessary because the code sniffer can also check text domains. By running phpcs via a Composer script instead, we can eliminate the Grunt dependency and all of its sub dependencies entirely, which reduces the problems associated with npm packages that frequently have vulnerabilities and need updating.

This also adds the phpcs-changed package and some related scripts (similar to the WooCommerce Core monorepo) so that the linter will only check file lines that have been modified instead of giving all code standard violations for the whole project.

Other changes:
* Upgrades Husky package to latest version and sets up a pre-commit hook for linting changes
* Adds a new npm `setup` script to simplify setting things up after cloning the repo
* Removes old Woo githooks that weren't being used
* Adds section to the readme about setting up for development

### How to test the changes in this Pull Request:

1. Make a fresh clone of this repo.
2. In the new clone, check out this branch.
3. If you use nvm, you can run `nvm use` to ensure you're on a compatible version of Node.
4. Run `npm run setup`. This should install all necessary npm and composer packages and set up the git hooks.
5. Open a PHP file in the repo and make a change that will cause a linting error.
6. Try to commit it. The pre-commit hook should run the linter and catch the error.
